src/pybind/mgr/cephadm/inventory.py

```diff
@@ -51,6 +51,8 @@ def assert_host(self, host: str) -> None:
             raise OrchestratorError('host %s does not exist' % host)
 
     def add_host(self, spec: HostSpec) -> None:
+        if spec.labels:
+            spec.labels = list(set((spec.labels)))
         self._inventory[spec.hostname] = spec.to_json()
```

Comment:
How do we deal with clusters that are already affected by this? Maybe we should change `HostSpec.{from,to}_json()´ instead?
.